flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecTemporalJoin.java

```diff
@@ -276,12 +278,21 @@ private TwoInputStreamOperator<RowData, RowData, RowData> createJoinOperator(
                     isLeftOuterJoin);
         } else {
             if (isTemporalFunctionJoin) {
-                return new TemporalProcessTimeJoinOperator(
-                        InternalTypeInfo.of(rightInputType),
-                        generatedJoinCondition,
-                        minRetentionTime,
-                        maxRetentionTime,
-                        isLeftOuterJoin);
+                if (config.get(ExecutionConfigOptions.TABLE_EXEC_ASYNC_STATE_ENABLED)) {
```

Comment:
As a heads up, changes like this should update consumedOptions for the JSON serialization.